src/main/java/io/aiven/kafka/connect/http/sender/AccessTokenHttpSender.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 Aiven Oy and http-connector-for-apache-kafka project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import static io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode.HEADER;
+import static io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode.URL;
+
+class AccessTokenHttpSender extends AbstractHttpSender implements HttpSender {
```

Comment:
I guess this logic only applies to OAuth2, so better to call it out in the name
```suggestion
class OAuth2AccessTokenHttpSender extends AbstractHttpSender implements HttpSender {
```